indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSource.java

```diff
@@ -123,13 +129,23 @@ CloseableIterator<InputEntity> createEntityIterator()
       private Iterator<OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType, RecordType>> recordIterator;
       private Iterator<? extends ByteEntity> bytesIterator;
       private volatile boolean closed;
+      private long createTime = System.currentTimeMillis();
+      private Long terminationTime = samplerConfigTimeoutMs != null ? createTime + samplerConfigTimeoutMs : null;
```

Comment:
I don't think you need to handle null here as `SamplerConfig.timeoutMs()` is already a primitive `int`.

Reply:
I had made it accept a null which was this was for, but changed this to check for 0 to be consistent with the way it's used elsewhere (to indicate no timeout specified)

Reply:
Changed it back to a nullable Integer to indicate no timeout, I think it's a clearer indicator than timeout<=0, and easier to add a deterministic test for the timeout (using a negative timeout value).

SamplerConfig.timeoutMs already had the convention of "0 or less indicates no timeout" so I kept that when constructing the RecordSupplierInputSource from the sampler spec.